cmd/controller/main.go

```diff
@@ -92,10 +96,20 @@ func main() {
 		HealthProbeBindAddress: fmt.Sprintf(":%d", opts.HealthProbePort),
 	})
 
+	cfg, err := config.New(ctx, clientSet, cmw)
+	if err != nil {
+		// this does not happen if the config map is missing or invalid, only if some other error occurs
+		logging.FromContext(ctx).Fatalf("unable to load config, %s", err)
+	}
+
+	if err := cmw.Start(ctx.Done()); err != nil {
+		logging.FromContext(ctx).Errorf("watching configmaps, config changes won't be applied immediately, %s", err)
```

Comment:
IMO we should panic here. Not a fan of karpenter running in a degraded state (until we have a usecase where customers don't want to give us access to configmaps).

Reply:
I could go either way, I figure it's unlikely to occur but in that case a degraded operation is still better than just failing to run.